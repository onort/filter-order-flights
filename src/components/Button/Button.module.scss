@import "../../styles/main.scss";

.button {
  cursor: pointer;
  border: none;
  border-radius: px-to-rem(2px);
  background-color: $grey4;
  color: $white;
  padding: map-get($spacers, small) map-get($spacers, x-large);

  &.primary {
    background-color: $primary;
    color: $white;
  }

  &.secondary {
    background-color: $secondary;
    color: $white;
  }

  .text {
    font-family: $font-family-primary;
    font-weight: 400;
  }
}
