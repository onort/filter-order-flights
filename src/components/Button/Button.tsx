import React from "react"
import cx from "classnames"

import styles from "./Button.module.scss"

type ButtonHTMLType = "submit" | "button" | "reset"

interface Props {
  className?: string
  onClick?: (e: React.MouseEvent<HTMLElement>) => void
  primary?: boolean
  secondary?: boolean
  text: string
  type?: ButtonHTMLType
}

const Button: React.SFC<Props> = props => {
  const { onClick, primary, secondary, text, type } = props
  const classNames = cx(
    styles.button,
    {
      [styles.primary]: primary,
      [styles.secondary]: secondary
    },
    props.className
  )
  return (
    <button className={classNames} type={type} onClick={onClick}>
      <span className={styles.text}>{text}</span>
    </button>
  )
}

Button.defaultProps = {
  type: "button"
}

export default Button
