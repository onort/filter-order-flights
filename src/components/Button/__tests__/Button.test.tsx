import React from "react"
import { shallow } from "enzyme"

import Button from "../"

const testText = "Test"

describe("<Button />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<Button text={testText} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have text rendered", () => {
    const wrapper = shallow(<Button text={testText} />)
    expect(wrapper.find("button").text()).toBe(testText)
  })

  it("should render primary button", () => {
    const wrapper = shallow(<Button text={testText} primary={true} />)
    expect(wrapper.find("button").hasClass("primary")).toBe(true)
  })

  it("should render secondary button", () => {
    const wrapper = shallow(<Button text={testText} secondary={true} />)
    expect(wrapper.find("button").hasClass("secondary")).toBe(true)
  })

  it("should pass button type correctly", () => {
    const wrapper = shallow(<Button text={testText} type="submit" />)
    expect(wrapper.find('button[type="submit"]').exists()).toBe(true)
  })

  it("should fire onClick when button is clicked", () => {
    const handleClick = jest.fn()
    const wrapper = shallow(<Button text={testText} onClick={handleClick} />)
    wrapper.find("button").simulate("click")
    expect(handleClick).toBeCalledTimes(1)
  })

  it("should be able to have a custom className", () => {
    const testClassName = "testClassName"
    const wrapper = shallow(
      <Button text={testText} className={testClassName} />
    )
    expect(wrapper.find("button").hasClass(testClassName)).toBe(true)
  })
})
