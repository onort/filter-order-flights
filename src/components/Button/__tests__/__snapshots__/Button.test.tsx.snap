// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    text="Test"
    type="button"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="text"
      >
        Test
      </span>,
      "className": "button",
      "onClick": undefined,
      "type": "button",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Test",
        "className": "text",
      },
      "ref": null,
      "rendered": "Test",
      "type": "span",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="text"
        >
          Test
        </span>,
        "className": "button",
        "onClick": undefined,
        "type": "button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test",
          "className": "text",
        },
        "ref": null,
        "rendered": "Test",
        "type": "span",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Button /> should match snapshot 2`] = `"<button class=\\"button\\" type=\\"button\\"><span class=\\"text\\">Test</span></button>"`;
